src/main/java/de/tum/in/www1/artemis/repository/PersistenceAuditEventRepository.java

```diff
@@ -52,6 +53,7 @@ default Page<PersistentAuditEvent> findAllWithDataByAuditEventDateBetween(Instan
             return Page.empty(pageable);
         }
         List<PersistentAuditEvent> result = findWithDataByIdIn(ids);
+        result.sort(Comparator.comparing(event -> ids.indexOf(event.getId())));
```

Comment:
```suggestion
        List<PersistentAuditEvent> result = findWithDataByIdInOrderByIdAsc(ids);
```
I don’t know exactly how the magic method name must be for it to work, but I think the sorting should be done already in the database since it is more efficient there.

Reply:
The problem might be that it's not ordering by id, but the method depends on the selection. The comparator uses the indexOf function to identify the position in the ordered list. I'm not sure how many magic methods we might need.

Reply:
Nevertheless, depending on the expected size of the ids list (Currently I guess it's 50), I think the performance might be OK even with the use of indexof. But tbh I'm unsure whether we should improve the performance by using a HashMap from ID to Position (that might be faster), because indexOf takes O(n) => O(n^2) for all element index lookups.

Reply:
Ah, I did not look at it correctly. My suggestion above won’t work, then, since it always sorts by date.
The user can select the column the results should be ordered by and the `Pageable` ensures the `ids` list are sorted correctly.

Reply:
Yeah I agree, the optimal way would be if there was only one clean database call. 
But since we need to sort by the returned id's, (which depend on the pageable) and there are only 50 elements at a time, I think this extra sorting step is ok from a performance and maintainability perspective.
If you have a cleaner sorting approach that's not O(n^2), I'd be happy to add it.

Reply:
For 50 elements I don't think that it's worth :D